### ISSUE TITLE
Unable to create the long name relations in PostgreSQL

### ISSUE THREAD

**Issue by ArtemHU** [Snowflake-Labs/dbt_constraints#19]
Since PostgreSQL has a predefined limit on the identifier length - 63 bytes, there is a chance that we run into the duplicate name issue. 
How to handle it( the possible solution):
by generating a unique constraint name -  https://github.com/Snowflake-Labs/dbt_constraints/pull/18

**Comment by danflippo** [comment]
First, thank you for identifying the issue and proposing a solution. 

I agree that we need to make a change. However, I would like to avoid creating dependencies with other dbt packages due to the versioning issues it can create for users. It is something I discussed with dbt-Labs before publishing the package. I originally was going to leverage the implementation of multi-column unique tests that dbt-Utils has. Instead, to make it more flexible for users, I created my own similar implementation.

I have tweaked your logic in the pull request to use the PostgreSQL md5() function, which is what dbt-utils would have generated as well. I also made the logic conditional so that it will create the original constraint names whenever they are 63 characters or less. I added a new model to the integration tests to test it using both paths.

Also, we don't need the execute test on your line 93 because it is already part of line 115 in create_constraints.sql. That is the first line of the macro that is invoked on-run-end to begin running dbt_constraints. If we are not executing, it immediately returns before going any further.

Finally, this problem is actually much worse for Oracle users because most Oracle DB have their identifier limit set to 30 characters and you couldn't even change it before Oracle 12c. I've added logic for the Oracle macros very similar to the logic we added for PostgreSQL.

Thanks again!
